llvm/utils/git/code-format-helper.py

```diff
@@ -31,9 +45,10 @@ def comment_tag(self) -> str:
     def instructions(self) -> str:
         raise NotImplementedError()
 
-    def format_run(
-        self, changed_files: list[str], args: argparse.Namespace
-    ) -> str | None:
+    def has_tool(self) -> bool:
+        raise NotImplementedError()
+
+    def format_run(self, changed_files: list[str], args: FormatArgs) -> str | None:
```

Comment:
What version of Python is required to run this hook? With `Python 3.9.6`, I'm getting this error:

```
  File "/Users/ldionne/code/llvm/.git/hooks/pre-commit", line 35, in <module>
    class FormatHelper:
  File "/Users/ldionne/code/llvm/.git/hooks/pre-commit", line 51, in FormatHelper
    def format_run(self, changed_files: list[str], args: FormatArgs) -> str | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

I think this is because `str | None` is not recognized as a type annotation in that version of Python?

Reply:
Hmm. Good point, I just used whatever was installed for me locally. I will try with the supported LLVM version to make sure.

Reply:
I tested this script with Python 3.6 now, so we should be good.